### PR TITLE
[FEATURE] Explain TeaController in Tutorial

### PR DESCRIPTION
**Pull request by linawolf**
releases: main, 11.5

### CODE REVIEW

File: Documentation/ExtensionArchitecture/Tutorials/Tea/Controller.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+==========
+Controller
+==========
+
+The controller controls the flow of data between the view and the
+data repository containing the model.
+
+A controller can contain one or more actions. Each of them is a method which
+ends on the name "Action" and returns an object of type
+:php:`Psr\Http\Message\ResponseInterface`.
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Tutorials/Tea/Controller.rst:16]:
```suggestion
:php:`\Psr\Http\Message\ResponseInterface`.
```

File: Documentation/ExtensionArchitecture/Tutorials/Tea/Index.rst [hunk 2]
```diff
@@ -49,6 +49,11 @@ Steps in this tutorial:
 
     The repository helps us to fetch tea objects from the database.
 
+#.  :ref:`The Controller <extbase_tutorial_tea_controller>`
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Tutorials/Tea/Index.rst:52]:
```suggestion
#.  :ref:`The controller <extbase_tutorial_tea_controller>`
```

File: Documentation/ExtensionArchitecture/Tutorials/Tea/Controller.rst [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+In the following action a tea object should be displayed in the view:
+
+.. include:: /CodeSnippets/Tutorials/Tea/Classes/Domain/Controller/ShowAction.rst.txt
+
+This action would be displayed if an URL like the following would be requested:
+:samp:`https://www.example.org/myfrontendplugin?tx_tea[action]=show&tx_tea[controller]=tea&tx_tea[tea]=42&chash=whatever`.
+
+So where does the model :php:`Tea $tea` come from? The only reference we had
+to the actual tea to be displayed was the id :php:`42`. In most cases the
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Tutorials/Tea/Controller.rst:26]:
```suggestion
to the actual tea to be displayed was the ID :php:`42`. In most cases, the
```

File: Documentation/ExtensionArchitecture/Tutorials/Tea/Controller.rst [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+parent class :php:`ActionController` will take care of matching parameters to
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Tutorials/Tea/Controller.rst:27]:
```suggestion
parent class :php:`\TYPO3\CMS\Extbase\Mvc\Controller\ActionController` will take care of matching parameters to
```

File: Documentation/ExtensionArchitecture/Tutorials/Tea/Controller.rst [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+objects or models. In more advanced scenarios it is necessary to influence
+the parameter matching. But in our scenario it is sufficient to know this
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Tutorials/Tea/Controller.rst:29]:
```suggestion
the parameter matching. But in our scenario it is sufficient to know that this
```

File: Documentation/ExtensionArchitecture/Tutorials/Tea/Controller.rst [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+happens automatically in the controller.
+
+The following action expects not parameters. It fetches all available tea
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Tutorials/Tea/Controller.rst:32]:
```suggestion
The following action expects no parameters. It fetches all available tea
```

File: Documentation/ExtensionArchitecture/Tutorials/Tea/Controller.rst [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+objects from the repository and hands them to the view:
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Tutorials/Tea/Controller.rst:33]:
```suggestion
objects from the repository and hands them over to the view:
```

File: Documentation/ExtensionArchitecture/Tutorials/Tea/Controller.rst [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+..  include:: /CodeSnippets/Tutorials/Tea/Classes/Domain/Controller/IndexAction.rst.txt
+
+The controller has to access the :php:`TeaRepository` to find all available tea
+objects. We use :ref:`Dependency Injection <DependencyInjection>` to make the
+repository available to the controller: The method :php:`injectTeaRepository()`
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Tutorials/Tea/Controller.rst:39]: The method "injectTeaRepository" is not available in the snippet above.

File: Documentation/ExtensionArchitecture/Tutorials/Tea/Controller.rst [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+will be called automatically with an initialized :php:`TeaRepository` when
+the :php:`TeaController` is created.
+
+Both action methods return a call to the function :php:`$this->htmlResponse()`.
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Tutorials/Tea/Controller.rst:43]:
```suggestion
Both action methods return a call to the method :php:`$this->htmlResponse()`.
```

File: Documentation/ExtensionArchitecture/Tutorials/Tea/Controller.rst [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+This method is implemented in the parent class :php:`ActionController` and is
+a shorthand method to create a response from the response factory and attach
+the rendered content. Let us have a look at what happens in this method:
+
+..  include:: /CodeSnippets/Tutorials/Tea/Classes/Domain/Controller/HtmlResponse.rst.txt
+
+You can also use this code directly in your controller if you need to return
+a different HTTP header. If a different rendering from the standard view is
+necessary you can just path the rendered HTML content to this method. There
+is also a short hand method for JSON called :php:`jsonResponse()`.
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Tutorials/Tea/Controller.rst:53]:
```suggestion
is also a shorthand method for returning JSON called :php:`jsonResponse()`.
```

File: Documentation/ExtensionArchitecture/Tutorials/Tea/Controller.rst [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+Both action methods return a call to the function :php:`$this->htmlResponse()`.
+This method is implemented in the parent class :php:`ActionController` and is
+a shorthand method to create a response from the response factory and attach
+the rendered content. Let us have a look at what happens in this method:
+
+..  include:: /CodeSnippets/Tutorials/Tea/Classes/Domain/Controller/HtmlResponse.rst.txt
+
+You can also use this code directly in your controller if you need to return
+a different HTTP header. If a different rendering from the standard view is
+necessary you can just path the rendered HTML content to this method. There
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Tutorials/Tea/Controller.rst:52]:
```suggestion
necessary you can just pass the rendered HTML content to this method. There
```

**Review by brotkrueml — CHANGES_REQUESTED** [review]

File: Documentation/ExtensionArchitecture/Tutorials/Tea/Controller.rst (hunk 8 above)
Comment (linawolf) [reply Documentation/ExtensionArchitecture/Tutorials/Tea/Controller.rst:39]: updated the snippet

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**